spec/workers/process_content_change_worker_spec.rb

```diff
@@ -55,5 +55,15 @@
 
       described_class.new.perform(processed_content.id)
     end
+
+    it "enforces job uniqueness with the correct sidekiq-unique-jobs option" do
+      expect do
+        SidekiqUniqueJobs.use_config(enabled: true) do
+          allow(SidekiqUniqueJobs.logger).to receive(:info)
+          expect(described_class).to receive(:uniqueness_with).with([content_change.id])
+          described_class.perform_async(content_change.id)
+        end
+      end.to output.to_stderr # silence redis-namespace warning output during the tests
+    end
```

Comment:
Nice one sussing this out. It looks rather thorny, I can see your code has picked up some wounds along the way.

I'd suggest doing a refactoring. Instead of defining this test in `#perform` it probably should be in `.perform_asnyc` with something like this:

```
  describe ".perform_async" do
    around do |example|
      SidekiqUniqueJobs.use_config(enabled: true, logger: Logger.new("/dev/null")) do
        # We're expecting redis-namespace to raise a warning unfortunately
        expect { example.run }.to output.to_stderr
      end
    end

    it "enforces job uniqueness with the correct sidekiq-unique-jobs option" do
      expect(described_class).to receive(:uniqueness_with).with([content_change.id])
      described_class.perform_async(content_change.id)
    end
  end
```

It does feel a bit wrong having an `expect {}.to output.to_stderr` but I don't know a better approach and at least that will actually tell us if we manage to resolve the warning by dropping redis namespace.

Reply:
I like it, very elegant. Thanks @kevindew

Reply:
@kevindew When running the entire test suite, I am finding one of the two tests fails, because the redis warning is output only once, the first time.

What do you think about using a non-namespaced redis db in tests? :

```
Sidekiq.configure_server do |config|
  config.redis = { db: Rails.env.test? ? 1 : 0 }
end

Sidekiq.configure_client do |config|
  config.redis = { db: Rails.env.test? ? 1 : 0 }
end
```

Reply:
> It does feel a bit wrong having an expect {}.to output.to_stderr but I don't know a better approach and at least that will actually tell us if we manage to resolve the warning by dropping redis namespace.

I guess that's a statement in favour of not using a redis db just for tests 🤔

Reply:
Oh darn, that's a pain.

This should sort it:

```
  describe ".perform_async" do
    around do |example|
      SidekiqUniqueJobs.use_config(enabled: true, logger: Logger.new("/dev/null")) do
        example.run
      end
    end

    # We're expecting redis-namespace to raise a warning unfortunately
    before { allow_any_instance_of(Redis::Namespace).to receive(:warn) }

    it "enforces job uniqueness with the correct sidekiq-unique-jobs option" do
      expect(described_class).to receive(:uniqueness_with).with([content_change.id])
      described_class.perform_async(content_change.id)
    end
  end
```

Reply:
Works a charms, thanks @kevindew